aws/identity/lib/policies.ts

```diff
@@ -1,28 +1,128 @@
+// limitations under the License.
+
 import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+import * as common from "./common";
 
-import * as config from "./config";
+type IamPolicyDocuments = "eksAdmins" | "useExistingIamRoles";
 
-const useExistingIamRolesPolicyDocument: aws.iam.PolicyDocument = {
-    Version: "2012-10-17",
-    Statement: [{ Effect: "Allow", Action: "iam:PassRole", Resource: "*" }]
+export const baselineIamPolicyDocuments: {
+    [PolicyDocument in IamPolicyDocuments]: aws.iam.PolicyDocument
+} = {
+    eksAdmins: {
+        Version: "2012-10-17",
+        Statement: [{ Effect: "Allow", Action: ["eks:*", "ec2:DescribeImages"], Resource: "*" }],
+    },
+    useExistingIamRoles: {
+        Version: "2012-10-17",
+        Statement: [{ Effect: "Allow", Action: "iam:PassRole", Resource: "*" }],
+    },
 };
-export const useExistingIamRoles =
-    config.defineUseExistingIamRolesPolicy == "true"
-        ? new aws.iam.Policy("iamPass", {
-              description:
-                  "Allows IAM Users to attach or modify existing IAM Roles to AWS resources",
-              policy: JSON.stringify(useExistingIamRolesPolicyDocument)
-          })
-        : null;
-
-const eksAdminPolicyDoc: aws.iam.PolicyDocument = {
-    Version: "2012-10-17",
-    Statement: [{ Effect: "Allow", Action: ["eks:*", "ec2:DescribeImages"], Resource: "*" }]
+
+// NOTE: `foo?: Bar` is not equivalent to `foo: Bar | undefined` class members. The former indicates
+// that the field is optional on the class, while the latter requires that it exists on the class,
+// and has a field that is `Bar | undefined`.
+type IBaselineIamPolicies = { [PolicyDocument in IamPolicyDocuments]: aws.iam.Policy | undefined };
+
+export type BaselinePolicyArgs = Pick<
+    aws.iam.AccessKeyArgs,
+    Exclude<keyof aws.iam.AccessKeyArgs, "policy">
+>;
+
+export type BaselineIamPoliciesOptions = {
+    [PolicyCreateOption in "defineEksAdminsPolicy" | "defineUseExistingIamRolesPolicy"]?:
+        | boolean
+        | BaselinePolicyArgs
 };
-export const eksAdmins =
-    config.defineEksAdminsPolicy == "true"
-        ? new aws.iam.Policy("eksAdmins", {
-              description: "Administrative access for EKS clusters",
-              policy: JSON.stringify(eksAdminPolicyDoc)
-          })
-        : null;
+
+export class BaselineIamPolicies extends pulumi.ComponentResource implements IBaselineIamPolicies {
+    public readonly eksAdmins: aws.iam.Policy | undefined;
+    public readonly useExistingIamRoles: aws.iam.Policy | undefined;
+
+    constructor(
+        name: string,
+        args?: BaselineIamPoliciesOptions,
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super(`${common.groupName}:index:BaselineIamPolicies`, name, args, opts);
+
+        args = args === undefined ? {} : args;
+
+        //
+        // IAM Policies that we create by default.
+        //
```

Comment:
this comment doesn't coincide with the code block right after it w.r.t. `args.defineUseExistingIamRolesPolicy`

Reply:
Hmm, I am pretty sure this is right, though perhaps confusing. Basically if `args.defineUseExistingIamRolesPolicy` is `undefined`, we want to set it to `true`; if not, just leave it how it is.

The effect of this is to default to `true`.

```
        args.defineUseExistingIamRolesPolicy =
            args.defineUseExistingIamRolesPolicy === undefined
                ? true
                : args.defineUseExistingIamRolesPolicy;
```

Reply:
Makes sense. I'm still getting used to TS, and the ternary makes me think a bit